File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = []
        curr = []
        n = len(nums)

        def helper(index,csum):

            if csum == target :
                ans.append(curr.copy())
                return

            if csum > target:
                return 
            
            for i  in range(index,n):

                curr.append(nums[i])

                helper(i,csum+nums[i])

                curr.pop()

        helper(0,0)

        return ans

            
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    // public int getMin(int nums, int l, int r){
    //     int m = (l+r)/2;
    //     if(nums[l]< nums[m]){
    //         return 

    //     }
    // }
    public int findMin(int[] nums) {
        int l=0, r= nums.length-1;
       
        while(l<r){
            int m = (l+r)/2;
            if(nums[m]< nums[r]){
                r = m;
            }else{
                l = m+1;
            }
        }
        return nums[l];
    }
}
